### PR TITLE
fix Test failed case from PR/18277

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: fix the failed test from #18277, because the error maybe come from `tikv` or `executor` module.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Release note 

- update test of PR/18277

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -5949,9 +5949,13 @@ func (s *testSuite) TestKillTableReader(c *C) {
 	tk.MustExec("create table t (a int)")
 	tk.MustExec("insert into t values (1),(2),(3)")
 	c.Assert(failpoint.Enable(retry, `return(true)`), IsNil)
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
 		c.Assert(int(errors.Cause(tk.QueryToErr("select * from t")).(*terror.Error).ToSQLError().Code), Equals, executor.ErrQueryInterrupted.Code())
+		wg.Done()
 	}()
 	time.Sleep(1 * time.Second)
```
Comment (SunRunAway) [inline executor/executor_test.go:5958]: Could this sleep be removed now?

**Comment by coocood** [comment]
LGTM

File: executor/executor_test.go (hunk 1 above)
Comment (fzhedu) [reply executor/executor_test.go:5958]: It trys to wait the `select` in the goroutne to run.

**Pull request merged**